Parsers/ASimRegistryEvent/Parsers/vimRegistryEventMicrosoftWindowsEvent.yaml

```diff
@@ -13,84 +13,144 @@ References:
\ No newline at end of file
+    let RegistryEvents=(){
+        let ASIM_GetWindowsUserType = (username:string, sid:string) { 
+            case ( 
+                username endswith "$" or sid in ("S-1-5-18", "S-1-5-19", "S-1-5-20"), "Machine"
+                , isempty(username) or sid == "S-1-0-0", ""
+                , "Regular"
+            )
+        };
+        let ASIM_GetAccountType = (sid:string) { 
+            iif ( 
+                sid in ("S-1-0-0","S-1-5-18", "S-1-5-19", "S-1-5-20"), "Simple"
+                , "Windows"
+            )
+        };
+        let ASIM_ParseWindowsEvents = (WindowsEvent:(EventData:dynamic)) {
+            WindowsEvent
+            | extend ActorUsername = iif(isnotempty(EventData.SubjectDomainName), strcat(EventData.SubjectDomainName, @'\', EventData.SubjectUserName), EventData.SubjectUserName)
+                , ActorDomainName = tostring(EventData.SubjectDomainName)
+                , ActorUserId = tostring(EventData.SubjectUserSid)
+                , ActorSessionId = tostring(EventData.SubjectLogonId)
+                , ActingProcessName = tostring(EventData.ProcessName)
+                , ActingProcessId = tostring(toint(tolong(EventData.ProcessId)))
+                , RegistryKey = iif(EventData.ObjectName startswith @"\REGISTRY\MACHINE", replace_string(tostring(EventData.ObjectName), @"\REGISTRY\MACHINE","HKEY_LOCAL_MACHINE")
+                    , replace_string(tostring(EventData.ObjectName), @"\REGISTRY\USER","HKEY_USERS")
+                )
+        };
+        let ASIM_ParseSecurityEvents = (SecurityEvent:(SubjectDomainName:string,SubjectUserName:string,ProcessId:string,ObjectName:string,SubjectUserSid:string,SubjectLogonId:string,ProcessName:string)) {
+            SecurityEvent
+            | project-rename ActorUsername = SubjectUserName
+                , ActorUserId = SubjectUserSid
+                , ActorSessionId = SubjectLogonId
+                , ActingProcessName = ProcessName
+                , ActorDomainName = SubjectDomainName
+            | extend ActorUsername = iif(isnotempty(ActorDomainName), strcat(ActorDomainName, @'\', ActorUsername), ActorUsername)
+                , ActingProcessId = tostring(toint(tolong(ProcessId)))
+                , RegistryKey = iif(ObjectName startswith @"\REGISTRY\MACHINE", replace_string(ObjectName, @"\REGISTRY\MACHINE","HKEY_LOCAL_MACHINE")
+                    , replace_string(ObjectName, @"\REGISTRY\USER","HKEY_USERS")
+                )
+        };
+        let Event4663TypeLookup = datatable (AccessMask:string,EventType:string)
+        [
+            "0x1", "RegistryValueRead"
+            , "0x10", "MetadataModified"
+            , "0x10000", "ObjectDeleted"
+            , "0x2", "RegistryValueSet"
+            , "0x20000", "MetadataAccessed"
+            , "0x20006", "MetadataAccessed"
+            , "0x40000", "MetadataModified"
+            , "0x8", "RegistrySubkeyEnumerated"
+            , "0x80000", "MetadataModified"
```

Comment:
Do you think it is worth adding specific value for owner change?

Reply:
I tried to reproduce owner changes but the events were not created in my environment